rdtools/analysis.py

```diff
@@ -177,9 +192,9 @@ def calc_clearsky_poa(self, times=None, rescale=True, **kwargs):
         if times is None:
             times = self.poa.index
         if self.pvlib_location is None:
-            raise ValueError('pvlib location must be provided')
+            raise Exception('pvlib location must be provided using set_clearsky()')
         if self.pv_tilt is None or self.pv_azimuth is None:
-            raise ValueError('pv_tilt and pv_azimuth must be provided')
+            raise Exception('pv_tilt and pv_azimuth must be provided using set_clearsky()')
```

Comment:
I'm not sure what exception type makes the most sense to raise here (and in `calc_clearsky_tamb` and `clearsky_preprocess`), but ideally we wouldn't be raising bare `Exception`s.  `ValueError` is a little awkward because it's raised in a different function from where the relevant argument is passed.  Maybe `AttributeError`?  Not sure.

Reply:
Maybe ValueError would be ok.  I'm looking through and not finding any great suggestions in the list of errors.  Here's description of ValueError, which I guess applies here :  
>exception ValueError
Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.

Reply:
NameError would also work here - we're expecting a named parameter pv_tilt and pv_azimuth that don't exist yet..